TreeTopView

duhan9836 opened the pull request:
You are given a pointer to the root of a binary tree. Print the top view of the binary tree.
Top view means when you look the tree from the top the nodes, what you will see will be called the top view of the tree.

duhan9836 committed:
TreeTopView

You are given a pointer to the root of a binary tree. Print the top view of the binary tree.
Top view means when you look the tree from the top the nodes, what you will see will be called the top view of the tree.

caitao1985 commented on treeTopView.py:
```diff
@@ -0,0 +1,86 @@
+    def __init__(self):
+        self.root = None
+
+    def create(self, val):
```
better renamed to `insert`, can you implement a recursive version?

duhan9836 replied:
the setup for class was given by the question. However, I'll think over the recursive implementation.

caitao1985 commented on treeTopView.py:
```diff
@@ -0,0 +1,86 @@
+    if nodelvl[0].left:
+        ret.append((nodelvl[0].left,nodelvl[1]-1))
+    if nodelvl[0].right:
+        ret.append((nodelvl[0].right,nodelvl[1]+1))
```
return

duhan9836 replied:
thanks

duhan9836 committed:
Update treeTopView.py

update the topView function:
1. add full information to res;
2. reorder the result as from left to right
update the children function:
add return
